src/Data/Auth/Providers/AuthContext.ts

```diff
@@ -0,0 +1,51 @@
+import { createContext } from "react";
+
+/**
+ * Interface for the authentication context properties.
+ */
+export interface GetAuthProvider {
+  /**
+   * Function to get the current user.
+   * @returns The current user, or undefined if no user is logged in.
+   */
+  getUser: () => string | undefined;
+
+  /**
+   * Function to log in a user.
+   */
+  login: () => void;
+
+  /**
+   * Function to log out the current user.
+   */
+  logout: () => void;
+
+  /**
+   * Function to update the current user and their token.
+   * @param user - The new user.
+   * @param token - The new token.
+   */
+  updateUser: (user: string, token: string) => void;
+
+  /**
+   * Function to get the current user's token.
+   * @returns The current user's token, or null if no user is logged in.
+   */
+  getToken: () => string | null;
+}
+
+/**
+ * The default values for the authentication context.
+ */
+export const defaultAuthContext: GetAuthProvider = {
```

Comment:
Why have a default provider when there is also the NoAuth provider?

Reply:
default context variable is the default values for the creation of the contexts, as well as in the noAuthProvider or test suites that's why it got separated from the `createContext()`

Reply:
Okay, but in that case you can simply import the context instead of reinitializing it by importing the defaultAuthContext.

Whether you have a specific object for it or handle it like so:

```ts
import { createContext } from "react";

interface  AuthProviderInterface {
  getUser: () => void;
  login: () => void;
  logout: () => void;
  updateUser: (user: string, token: string) => void;
  getToken: () => string | null;
}

export const AuthContext= createContext<AuthProviderInterface >({
  getUser: () => undefined,
  login: () => {},
  logout: () => {},
  updateUser: (_user: string, _token: string) => {},
  getToken: () => null, // or the return "no token"; if you consider the default being without authentication.  although I'm not certain that would be needed, or make sense to return "no token" instead of null
});
```

You should be able to use it like this since it is already instanciated. You don't need to reset the defaults or make sure it has defaults because it already has it built in when you declared it. 
```ts
  const authContext= useContext(AuthContext);
  // rest of code
  return (
     <AuthContext.Provider value={authContext}>
         {children}
     </AuthContext.Provider>
  )
```

Reply:
Oh, didn't think of that, it felt wrong to call context in the component that will provide it to the application. variable itself will stay as it's required in the test files

Reply:
Yes, then you can make a ticket and reference this PR//Comment.